File: frontend/src/types/sales.ts
export interface SalesSummary {
  id: number
  department_id: string
  date: string
  total_sales: number          // прайс (DishSumInt)
  total_paid: number | null    // к оплате (DishDiscountSumInt); null до бэкфилла
  created_at: string
  updated_at: string
  synced_at: string
}

export interface SalesByHour {
  id: number
  department_id: string
  date: string
  hour: number
  sales_amount: number         // прайс
  paid_amount: number | null   // к оплате; null до бэкфилла
  created_at: string
  updated_at: string
  synced_at: string
}

export interface SalesHeatmap {
  from_date: string | null
  to_date: string | null
  department_id: string | null
  /** 7×24 grid: rows are Mon..Sun, cols are 0..23 hours. */
  grid: number[][]
  /** 7×24 grid «к оплате» (paid). */
  grid_paid: number[][]
}

export interface SyncResult {
  status: 'success' | 'error'
  message: string
  from_date: string
  to_date: string
  summary_records?: number
  hourly_records?: number
  total_raw_records?: number
  details?: string
  error_type?: string
}

File: app/models/sales.py
from sqlalchemy import Column, String, ForeignKey, DateTime, Float, Integer, Date
from sqlalchemy.orm import relationship
from sqlalchemy.dialects.postgresql import UUID
from datetime import datetime
from ..db import Base


class SalesSummary(Base):
    __tablename__ = "sales_summary"

    id = Column(Integer, primary_key=True, index=True)
    department_id = Column(UUID(as_uuid=True), ForeignKey("departments.id"), nullable=False, index=True)
    date = Column(Date, nullable=False, index=True)
    total_sales = Column(Float, nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    synced_at = Column(DateTime, default=datetime.utcnow)

    department = relationship("Department")

    __table_args__ = (
        {"schema": None},
    )


class SalesByHour(Base):
    __tablename__ = "sales_by_hour"

    id = Column(Integer, primary_key=True, index=True)
    department_id = Column(UUID(as_uuid=True), ForeignKey("departments.id"), nullable=False, index=True)
    date = Column(Date, nullable=False, index=True)
    hour = Column(Integer, nullable=False, index=True)
    sales_amount = Column(Float, nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    synced_at = Column(DateTime, default=datetime.utcnow)

    department = relationship("Department")

    __table_args__ = (
        {"schema": None},
    )


class AutoSyncLog(Base):
    __tablename__ = "auto_sync_log"

    id = Column(Integer, primary_key=True, index=True)
    sync_date = Column(Date, nullable=False, index=True)
    sync_type = Column(String(50), nullable=False)
    status = Column(String(20), nullable=False, index=True)
    message = Column(String(500), nullable=True)
    summary_records = Column(Integer, default=0)
    hourly_records = Column(Integer, default=0)
    total_raw_records = Column(Integer, default=0)
    error_details = Column(String(1000), nullable=True)
    executed_at = Column(DateTime, default=datetime.utcnow, index=True)
    created_at = Column(DateTime, default=datetime.utcnow)
